src/index.js

```diff
@@ -50,20 +54,55 @@ const hunt = () => {
   return enemy.msgs.encounter;
 };
 
+/**
+ * @param {number} low The lower bound (inclusive)
+ * @param {number} high The upper bound (exclusive)
+ * @returns {number} A random number between low and high
+ */
+const randomRange = (low, high) => Math.random() * (high - low) + low;
+
+/**
+ * @param {number} low The lower bound (inclusive)
+ * @param {number} high The upper bound (inclusive)
+ * @returns {number} A random integer between low and high
+ */
+const irandomRange = (low, high) => Math.floor(randomRange(low, high + 1));
+
+/**
+ * @returns {number} a random number between 0.8 and 1.2
+ */
+const random20 = () => 1 + irandomRange(-20, 20) / 100;
+
+/**
+ * @param {number[]} list A list of numbers
+ * @returns {number} the sum of numbers in the list
+ */
+const sum = (list) => list.reduce((total, value) => total + value, 0);
+
+/**
+ * @param {Map<T, number>} options A map of options to their weight
+ * @returns {T} One of the options, chosen randomly
+ */
+const chooseWeighted = (options) => {
+  const totalWeight = sum([...options.values()]);
+  let which = irandomRange(1, totalWeight);
+  for (const [item, weight] of options) {
+    which -= weight;
+    if (which <= 0) { return item; }
+  }
+};
+
```

Comment:
Ooh ok

`.values()` is a method of the `Map` class. A map is just an association of `key: value`, so `.values()` will just return the value parts in order, discarding the keys.

The dots is "spread", which just takes a thing and spreads it out into whatever context you are in:
```javascript
// you can spread an array into function arguments like this:
const list = [1, 2, 3];
function add(a, b, c) { return a + b + c; }
add(...list); // 6

// you can spread arrays into other arrays, similar to what was going on with values above
const list2 = [4, 5, 6];
const list3 = [...list, list2]; // [1, 2, 3, 4, 5, 6]

// you can spread objects into other objects
const x = { x: 1 };
const y = { y: 2 };
const xy = { ...x, ...y }; // { x: 1, y: 2 }
```

`.values)_` is kind of special in that it doesn't return an array, but what's called an "iterable". You can't do much with iterables other than spread it or you can `for (const item of iterable)` it. Often the only thing you want to do with an iterable is to turn it into an array, so that's what happens here with `[...options.values()]`.